inspirehep/modules/workflows/views/callback.py

```diff
@@ -99,26 +115,51 @@ def webcoll_callback():
     recids field.
     """
     recids = dict(request.form).get('recids', [])
-    pending_records = current_cache.get("pending_records") or dict()
-    for rid in recids:
-        if rid in pending_records:
-            objectid = pending_records[rid]
-            workflow_object = workflow_object_class.get(objectid)
-            base_url = _get_base_url()
-            workflow_object.extra_data['url'] = join(
-                base_url, 'record', str(rid)
+    results = {}
+    for recid in recids:
+        recid = int(recid)
+        if recid in results:
+            current_app.logger.warning('Received duplicated recid: %s', recid)
```

Comment:
I guess its fine to have a `warning` here. Not sure when a duplicate recid would be sent (could be a bug on legacy code if it happens).